software/controller/lib/sensors/sensor_base.h

```diff
@@ -15,20 +15,21 @@ limitations under the License.
 
 #pragma once
 
-#include "hal.h"
+#include "adc.h"
 #include "units.h"
 #include "vars.h"
 
 class AnalogSensor {
  public:
-  AnalogSensor(const char *name, const char *help_supplement, AnalogPin pin);
+  AnalogSensor(const char *name, const char *help_supplement, const GPIO::AdcChannel &channel,
```

Comment:
Might also be a good time to split `AnalogSensor` into separate file so that the `base` avoids the `adc` dependency and is mockable cheaper. And then I can also smell an i2c sensor class coming soon...

Reply:
done